src/Jaeger.Core/Util/DateTimeExtensions.cs

```diff
@@ -2,7 +2,7 @@
 
 namespace Jaeger.Core.Util
 {
-    public static class DateTimeExtensionMethods
+    public static class DateTimeExtensions
```

Comment:
I would prefer to use `DateTimeOffset` in the whole code base instead of mixing `DateTimeOffset` with `DateTime`. The code base looks like it uses Utc internally correct but still, it's unnecessary and bears a lot of bug potential (and needs this unnecessary extension class).

If you are fine with it, I could also submit the code for it. Especially, the `ToUnixTimeMilliseconds` method of `DateTimeOffset` uses UTC, independant of whatever timezone the user created the time for.

Reply:
Ah, seems like I missed that issue completely. Thanks also for providing the benchmark over there! Pretty interesting that it's that much slower compared to simple `DateTime` instances.